scene/resources/animated_texture.cpp

```diff
@@ -270,13 +270,18 @@ void AnimatedTexture::_bind_methods() {
 	BIND_CONSTANT(MAX_FRAMES);
 }
 
+void AnimatedTexture::_finish_non_thread_safe_setup() {
+	RenderingServer::get_singleton()->connect("frame_pre_draw", callable_mp(this, &AnimatedTexture::_update_proxy));
+}
+
 AnimatedTexture::AnimatedTexture() {
 	//proxy = RS::get_singleton()->texture_create();
 	proxy_ph = RS::get_singleton()->texture_2d_placeholder_create();
 	proxy = RS::get_singleton()->texture_proxy_create(proxy_ph);
 
 	RenderingServer::get_singleton()->texture_set_force_redraw_if_visible(proxy, true);
-	RenderingServer::get_singleton()->connect("frame_pre_draw", callable_mp(this, &AnimatedTexture::_update_proxy));
+
+	MessageQueue::get_main_singleton()->push_callable(callable_mp(this, &AnimatedTexture::_finish_non_thread_safe_setup));
```

Comment:
Wouldn't using the `CONNECT_DEFERRED` flag achieve something similar without having to introduce a new method?

Reply:
Good question, but this is another case of the difference between the main call queue and the thread override call queue is relevant. `CONNECT_DEFERRED` would use `MessageQueue::get_singleton()`, which during threaded loading wouldn't be the good old default `MessageQueue`, but the override the `ResourceLoader` has set for the loading thread. We need to bypass it.

Maybe we could consider making that a bit more transparent. However, as I see things, we are in a _de facto_ stage of colllection of information about the needs of resources dealing with a multi-threaded world and we're not ready yet to come up with solid API designs that are better than the manual approaches.